application/Synchronization/SynchronizerBase.cs

```diff
@@ -154,6 +134,9 @@ public void Dispose(bool disposing)
             {
                 //cleanup
                 Stop();
+                _amazonClient = null;
+                _watch = null;
+                //_telemetryClient.Flush();
```

Comment:
Why is this commented out?  If it is unnecessary, can it be removed instead?  If it is necessary, should it be uncommented?